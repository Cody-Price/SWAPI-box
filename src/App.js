import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
      randomCrawl: ''
    }
  }

  fetchFilms = async (filmsUrl) => {
    const films = await fetch(filmsUrl);
    const response = await films.json();
    const randomIndex = Math.floor(Math.random() * 7);
    const result = response.results[randomIndex].opening_crawl;
    return result;
  }

  async componentDidMount() {
    const url = 'https://swapi.co/api/'
    const swapiFetch = await fetch(url);
    const response = await swapiFetch.json();
    const randomCrawl = await this.fetchFilms(response.films);
    this.setState({randomCrawl});
  }

  render() {
    return (
      <div className="App">
        
      </div>
    );
  }
}

export default App;
