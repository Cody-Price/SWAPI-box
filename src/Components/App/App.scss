@import url('https://fonts.googleapis.com/css?family=Russo+One');

.App {
  height: 100%;
  font-family: 'Russo One', arial, sans-serif;
  width: 100%;
}