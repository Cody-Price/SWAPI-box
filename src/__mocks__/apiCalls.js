export const getRandomFilmCrawl = jest.fn().mockImplementation(() => {
  return Promise.resolve('Star Wars is awesome!')
})