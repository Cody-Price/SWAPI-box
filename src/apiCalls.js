import {
  filmCleaner,
  cardCleaner
} from './Helper.js';

export const fetchData = async (url) => {
  try {
    const response = await fetch(url)
    const data = await response.json()
    return data;
  } catch (error) {
    return error.message;
  }
}
export const getRandomFilmCrawl = async (filmUrl) => {
  const films = await fetchData(filmUrl)
  const randomCrawl = filmCleaner(films)
  return randomCrawl
}

export const fetchByMenu = async (selection) => {
  const url = `https://swapi.co/api/${selection}`
  const data = await fetchData(url);
  const results = cardCleaner(data.results, selection)
  return results;
}

export const fetchProperty = async (url) => {
  let returnedValue;
  if (typeof url === 'string') {
    returnedValue = await fetchData(url);
  } else {
    returnedValue = url.map(async address => {
      const property = await fetchData(address)
      return property.name
    })
  }
  return returnedValue;
}