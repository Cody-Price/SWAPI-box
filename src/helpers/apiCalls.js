import {
  filmCleaner,
  cardCleaner
} from './Cleaner.js';
import { fetchData } from './fetchData';

export const getRandomFilmCrawl = async (filmUrl) => {
  const films = await fetchData(filmUrl)
  const randomCrawl = filmCleaner(films)
  return randomCrawl
}

export const fetchByMenu = async (selection) => {
  const url = `https://swapi.co/api/${selection}`
  const data = await fetchData(url);
  const results = cardCleaner(data.results, selection)
  return results;
}

export const fetchProperty = async (url) => {
  let returnedValue;
  if (typeof url === 'string') {
    returnedValue = await fetchData(url);
  } else {
    returnedValue = url.map(async address => {
      const property = await fetchData(address)
      return property.name
    })
  }
  return returnedValue;
}