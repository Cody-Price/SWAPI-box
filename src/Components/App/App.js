import React, { Component } from 'react';
import './App.scss';
import ScrollingText from '../ScrollingText/ScrollingText.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      randomCrawl: '', 
      errorMessage: ''
    }
  }

  fetchFilms = async (filmsUrl) => {
    try {
      const films = await fetch(filmsUrl);
      console.log('hi')
    const response = await films.json();
    const randomIndex = Math.floor(Math.random() * 8);
    const result = {
      crawl: response.results[randomIndex].opening_crawl,
      title: response.results[randomIndex].title,
      episode: response.results[randomIndex].episode_id
    }
    return result
    } catch(error) {
    this.setState({
      errorMessage: error.message
      })
    } 
  }

  async componentDidMount() {
    try {
    const url = 'https://swapi.co/api/'
    const swapiFetch = await fetch(url);
    const response = await swapiFetch.json();
    const randomCrawl = await this.fetchFilms(response.films);
    this.setState({randomCrawl});
    } catch(error) {
      this.setState({
        errorMessage: error.message
      })
    }
  }

  render() {
    const { randomCrawl, title, episode } = this.state.randomCrawl;
    return (
      <div className="App">
        <ScrollingText title={title}
                        crawl={randomCrawl}
                        episode={episode} />
      </div>
    );
  }
}

export default App;
