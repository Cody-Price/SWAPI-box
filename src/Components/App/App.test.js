import React from 'react';
import App from './App';
import {shallow} from 'enzyme'; 
import * as API from '../../apiCalls.js'
import mockData from '../../mockData';
jest.mock('../../apiCalls.js')



describe('App', () => {
  let mockRandomCrawl
  let wrapper
  let mockUrl
  let mockResponse
  let mockSwapiFetch

  it('should match snapshot', () => {
    wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot() 
  })

  describe('componentDidMount', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<App />)
      mockUrl = 'https://swapi.co/api/'
      mockResponse = {
        "people": "https://swapi.co/api/people/",
        "planets": "https://swapi.co/api/planets/",
        "films": "https://swapi.co/api/films/",
        "species": "https://swapi.co/api/species/",
        "vehicles": "https://swapi.co/api/vehicles/",
        "starships": "https://swapi.co/api/starships/"
      }

    })

    it('calls API.getRandomFilmCrawl with the correct params', async () => {
      const expected = 'https://swapi.co/api/films'

      await wrapper.instance().componentDidMount()

      expect(API.getRandomFilmCrawl).toHaveBeenCalledWith(expected)
    })

    it('should update randomCrawl in state after our fetch call has been made successfully', async () => {
      const wrapper = shallow(<App />)
     
      const expectedState = {
      randomCrawl: "Star wars is awesome", 
      errorMessage: '',
      menuSelection: '',
      favorites: [],
      selectedCards: [],
      people: [],
      vehicles: [],
      planets: [],
      loadingStatus: false
      }
      
      wrapper.instance().componentDidMount(() => {
        expect(wrapper.state()).toEqual(expectedState)
      })

    })

    it('should set an error message if our fetch fails', async () => {      
      const expectedState = {
        randomCrawl: '', 
        errorMessage: 'Could not fetch',
        menuSelection: '',
        favorites: [],
        selectedCards: [],
        people: [],
        vehicles: [],
        planets: [],
        loadingStatus: false
      }

      wrapper.instance().componentDidMount(() => {
        expect(wrapper.state()).toEqual(expectedState)
      })
    })
  })
  describe('menuSelect', () => {
    let mockInitialState
    beforeEach(() => {
      mockInitialState = {
        randomCrawl: '', 
        errorMessage: '',
        menuSelection: '',
        favorites: [],
        selectedCards: [],
        people: [],
        vehicles: [],
        planets: [],
        loadingStatus: false
      }
    })
    it('should take a string of the users selection as an argument', () => {
      const wrapper = await shallow(<App/>)
      const mockSelection = 'people'
      
       wrapper.instance().menuSelect(mockSelection, () => {
       expect(wrapper.state('menuSelection')).toEqual(mockSelection)
      })
    })
    it.skip('should display an error message if there are no favorites saved and favorites have been selected', () => {
      const wrapper = shallow(<App/>)
      const mockSelection = 'favorites'

      wrapper.instance().menuSelect(mockSelection, () => {
        expect(mock)
      })
    })
  })
})

