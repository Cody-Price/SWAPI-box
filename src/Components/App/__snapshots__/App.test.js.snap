// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="twinkle"
        />,
        <Menu
          favorites={Array []}
          menuSelect={[Function]}
        />,
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/people"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/vehicles"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/planets"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "twinkle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "favorites": Array [],
          "menuSelect": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/people"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/vehicles"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/planets"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/people",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/vehicles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/planets",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="twinkle"
          />,
          <Menu
            favorites={Array []}
            menuSelect={[Function]}
          />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/people"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/vehicles"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/planets"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "twinkle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "favorites": Array [],
            "menuSelect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/people"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/vehicles"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/planets"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/people",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/vehicles",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/planets",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
